modules/xml/test/src/smithy4s/xml/XmlDecodingSpec.scala

```diff
@@ -0,0 +1,318 @@
+package smithy4s.xml
+
+import smithy4s.xml.internals.XmlSchemaVisitor
+import smithy4s.xml.internals.XmlCursor
+import weaver._
+import fs2._
+import fs2.data.xml._
+import fs2.data.xml.dom._
+import smithy4s.schema.Schema
+import cats.effect.IO
+import cats.syntax.all._
+import smithy4s.schema.Schema._
+import smithy.api.XmlName
+import smithy.api.XmlAttribute
+import smithy.api.XmlFlattened
+import smithy4s.ByteArray
+
+object XmlDecodingSpec extends SimpleIOSuite {
+
+  test("int") {
+    implicit val schema: Schema[Int] = int
+    val xml = "<x>1</x>"
```

Comment:
probably worth mentioning that anything is allowed in place of `x` for decoding - maybe let's stick to `default`? It could be the way we encode things by default too, and might make for a more reasonable value to use for reduced confusion (e.g. wtf is `x`?)

Reply:
maybe even `placeholder`, which would at least be consistent with what we do for SchemaId in some schemas